langlib/lang.runtime/src/main/java/org/ballerinalang/langlib/runtime/Sleep.java

```diff
@@ -35,13 +35,23 @@
 public class Sleep {
 
     private static final int CORE_THREAD_POOL_SIZE = 1;
+    private static final BigDecimal LONG_MAX = new BigDecimal(Long.MAX_VALUE);
 
     private static final ScheduledExecutorService executor = Executors.newScheduledThreadPool(CORE_THREAD_POOL_SIZE);
 
     public static void sleep(Environment env, BDecimal delaySeconds) {
+        BigDecimal delayDecimal = delaySeconds.decimalValue().multiply(new BigDecimal("1000.0"));
+        if (delayDecimal.compareTo(BigDecimal.ZERO) <= 0) {
```

Comment:
Can we direclty comapare delaySeconds.decimalValue() instead of after multiple with 1000?